### PR TITLE
[SPARK-39948][BUILD] Exclude hive-vector-code-gen dependency

### PR DESCRIPTION
**Pull request by zhouyifan279**
### What changes were proposed in this pull request?
Remove hive-vector-code-gen and its dependent jars from spark distribution

### Why are the changes needed?
hive-vector-code-gen is not used in spark 

Remove it to avoid vulnerability scanners's alert.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Passed current test cases

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @sunchao FYI

**Comment by LuciferYang** [comment]
@zhouyifan279 Title should be `[SPARK-39948][BUILD] Exclude...`

**Comment by zhouyifan279** [comment]
> @zhouyifan279 Title should be `[SPARK-39948][BUILD] Exclude...`

Thanks, corrected.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. (Pending CIs)

**Review by dongjoon-hyun — COMMENTED** [review]
BTW, @zhouyifan279 . The GitHub Action is not triggerred at your repository still.
Screenshot 2022-12-08 at 12 58 25 AM

**Comment by zhouyifan279** [comment]
> BTW, @zhouyifan279 . The GitHub Action is not triggerred at your repository still. Screenshot 2022-12-08 at 12 58 25 AM

@dongjoon-hyun thanks for your reminding. Several workflows were triggered by previous commits. I have cancelled them.

**Comment by srowen** [comment]
If we can exclude this code, then, the CVE doesn't affect Spark to begin with right?

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/package.scala [hunk 1]
```diff
@@ -117,7 +118,8 @@ package object client {
         "org.apache.derby:derby:10.14.1.0"),
       exclusions = Seq("org.apache.calcite:calcite-druid",
         "org.apache.curator:*",
-        "org.pentaho:pentaho-aggdesigner-algorithm"))
+        "org.pentaho:pentaho-aggdesigner-algorithm",
+        "org.apache.hive:hive-vector-code-gen"))
```
Comment (sunchao) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/package.scala:122]: do other Hive versions like `v2_2` have the same issue?

**Review by sunchao — APPROVED** [review]
LGTM

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/package.scala (hunk 1 above)
Comment (dongjoon-hyun) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/client/package.scala:122]: @zhouyifan279 Could you answer @sunchao 's question?

**Review by HyukjinKwon — APPROVED** [review]

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/package.scala (hunk 1 above)
Comment (dongjoon-hyun) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/client/package.scala:122]: Thank you for confirming, @zhouyifan279 .

**Comment by dongjoon-hyun** [comment]
Merged to master branch for Apache Spark 3.4.0.

**Comment by dongjoon-hyun** [comment]
What is your JIRA ID, @zhouyifan279 ?

**Comment by dongjoon-hyun** [comment]
I found your JIRA ID, `zhouyifan279`, and added it to Apache Spark contributor group.

Also, from the commit log, I found the following three JIRAs and assigned them to you.
- SPARK-39948
- SPARK-8731
- SPARK-37863

Thank you for your contribution, @zhouyifan279 .

**Comment by zhouyifan279** [comment]
> I found your JIRA ID, `zhouyifan279`, and added it to Apache Spark contributor group.
> 
> Also, from the commit log, I found the following three JIRAs and assigned them to you.
> 
> * [SPARK-39948](https://issues.apache.org/jira/browse/SPARK-39948)
> * [SPARK-8731](https://issues.apache.org/jira/browse/SPARK-8731)
> * [SPARK-37863](https://issues.apache.org/jira/browse/SPARK-37863)
> 
> Thank you for your contribution, @zhouyifan279 .

Thanks. It's my pleasure to do something for the Spark community.

**Pull request closed without merging**